app/src/js/server/redis_store.ts

```diff
@@ -103,8 +92,20 @@ export class RedisStore {
   /**
    * Wrapper for get
    */
-  public async get (key: string) {
-    return this.client.get(key)
+  public async get (key: string): Promise<string | null> {
```

Comment:
This seems to be a duplicate of the ProjectStore.loadState function, which checks redid first then memory. Was that not working?

Reply:
The purposes are different. If it is a cache miss, the data should be reloaded from storage and put in cache. This is the low-level cache behavior. `ProjectStore` can handle the other cases: 1) cache is not used 2) the state has not been created. They can both handle task state cache misses, but `RedisStore` is supposed to be independent and has the complete caching capability, since we may use it to cache other information outside project store such as user info.

Reply:
Down the road, it would be better to have a general interface for the redis store, so that it be easily used for the other purposes as a message broker and storage cache.

Reply:
I see. At the moment, the functionality is separate. `redis_store` is for caching, and `redis_pub_sub` is for message broker. Both rely on `redis_client`, which is the low-level interface to redis.

Reply:
That separation is great!